### PR TITLE
fix: changed query to accept user input in prepared sql statement

### PR DESCRIPTION
**Pull request by deepakrai9185720**
# Description
Github reported few vulnerability. Mentioning the slack conversation:
https://rudderlabs.slack.com/archives/C049T5P2Z97/p1667507291386099

## Notion Ticket
https://www.notion.so/rudderstacks/SQL-injection-vulnerability-in-TriggerWHUploads-rudder-server-25b7e45711034ad7889f458ad24768b0
## Security

- [x] The code changed/added as part of this pull request won't create any security issues with how the software is being used.

### CODE REVIEW

File: warehouse/warehouse.go [hunk 1]
```diff
@@ -1831,17 +1834,20 @@ func getPendingStagingFileCount(sourceOrDestId string, isSourceId bool) (fileCou
...
-	err = dbHandle.QueryRow(sqlStatement).Scan(&fileCount)
+	preparedSqlStatement, err = dbHandle.Prepare(sqlStatement)
+	if err != nil {
+		err = fmt.Errorf("query: %s preparation failed with Error: %w", strings.ReplaceAll(sqlStatement, "$1", sourceOrDestId), err)
+		return
+	}
+	err = preparedSqlStatement.QueryRow(sourceOrDestId).Scan(&fileCount)
 	if err != nil && err != sql.ErrNoRows {
-		err = fmt.Errorf("query: %s failed with Error : %w", sqlStatement, err)
+		err = fmt.Errorf("query: %s run failed with Error : %w", sqlStatement, err)
```
Comment (lvrach) [inline warehouse/warehouse.go:1850]: If you are not planning to reuse a prepared statement, then it is not necessary to do it in two steps.

We can simply do `dbHandle.QueryRow(sqlStatement, sourceOrDestId).Scan(&fileCount)`

**Review by lvrach — APPROVED** [review]
With some minor comments

File: warehouse/warehouse.go [hunk 2]
```diff
@@ -1807,16 +1807,19 @@ func getPendingStagingFileCount(sourceOrDestId string, isSourceId bool) (fileCou
 		FROM
 		  %[1]s
 		WHERE
-		  %[1]s.%[3]s = '%[2]s';
+		  %[1]s.%[2]s = $1;
```
[the remark this answers is not in the document]
Comment (deepakrai9185720) [reply warehouse/warehouse.go]: Done

File: warehouse/warehouse.go (hunk 1 above)
Comment (deepakrai9185720) [reply warehouse/warehouse.go]: Done

File: warehouse/warehouse.go [hunk 3]
```diff
@@ -1807,16 +1807,19 @@ func getPendingStagingFileCount(sourceOrDestId string, isSourceId bool) (fileCou
...
 `,
 		warehouseutils.WarehouseUploadsTable,
-		sourceOrDestId,
 		sourceOrDestColumn,
```
[the remark this answers is not in the document]
Comment (deepakrai9185720) [reply warehouse/warehouse.go:1814]: Done for the said variable. Doing it for the whole code will take time. Should we do it after as soon as get some bandwidth? Wdyt?

File: warehouse/warehouse.go [hunk 4]
```diff
@@ -1830,18 +1829,16 @@ func getPendingStagingFileCount(sourceOrDestId string, isSourceId bool) (fileCou
...
+		  AND %[3]s = $1;
 `,
 		warehouseutils.WarehouseStagingFilesTable,
 		lastStagingFileID,
-		sourceOrDestId,
 		sourceOrDestColumn,
 	)
-
-	err = dbHandle.QueryRow(sqlStatement).Scan(&fileCount)
+	err = dbHandle.QueryRow(sourceOrDestId, sourceOrDestId).Scan(&fileCount)
```
Comment (achettyiitr) [inline warehouse/warehouse.go:1839]:
```suggestion
	err = dbHandle.QueryRow(sqlStatement, sourceOrDestId).Scan(&fileCount)
```

**Review by achettyiitr — CHANGES_REQUESTED** [review]

**Review by achettyiitr — APPROVED** [review]

File: warehouse/warehouse.go (hunk 3 above)
[the remark this answers is not in the document]
Comment (lvrach) [reply warehouse/warehouse.go:1814]: Yeah, let's discuss further refactoring and prioritise accordingly.

**Pull request merged**